sqlglot/helper.py

```diff
@@ -466,6 +466,11 @@ def merge_ranges(ranges: t.List[t.Tuple[A, A]]) -> t.List[t.Tuple[A, A]]:
     return merged
 
 
+def filter_none(iterable: t.Iterable) -> t.Iterator:
```

Comment:
does this hurt performance?

Reply:
I can check, should I just run bench.py w/ before & after?

Reply:
I'd expect it to be a minimal hit (if any) due to the function call overhead but we can test it.

Reply:
function call overhead is a good amount

Reply:
i'm a bit wary of this helper now and think we may not need it

Reply:
FWIW, ran the benchmark with Python 3.10.13 and got these (main vs my branch, respectively):

```
➜  benchmarks git:(main) python bench.py
|           Query |         sqlglot |         sqltree |
| --------------- | --------------- | --------------- |
|            tpch |   0.01261 (1.0) | 0.00986 (0.782) |
|           short |   0.00067 (1.0) | 0.00056 (0.841) |
|            long |   0.01010 (1.0) | 0.00721 (0.713) |
|           crazy |   0.03203 (1.0) | 0.02216 (0.691) |
➜  benchmarks git:(jo/array_join_improvement) python bench.py
|           Query |         sqlglot |         sqltree |
| --------------- | --------------- | --------------- |
|            tpch |   0.01225 (1.0) | 0.00792 (0.646) |
|           short |   0.00066 (1.0) | 0.00056 (0.846) |
|            long |   0.01212 (1.0) | 0.00729 (0.601) |
|           crazy |   0.03215 (1.0) | 0.02237 (0.695) |
```

The test suite run times (`make style && make test`):

main: Ran 991 tests in 97.591s
my branch: Ran 991 tests in 98.233s

Reply:
```python
from timeit import timeit

data = [1, 2, 3, None]
data_big = data * 100

def filter_none(iterable):
    return (x for x in iterable if x is not None)

setup = "from __main__ import data, data_big, filter_none"

print("comprehension:", timeit("[a for a in data if a is not None]", setup=setup))
print("function:", timeit("[a for a in filter_none(data)]", setup=setup))
print("comprehension (big):", timeit("[a for a in data_big if a is not None]", setup=setup))
print("function (big):", timeit("[a for a in filter_none(data_big)]", setup=setup))
```

Output:
```
comprehension: 0.2941456299740821
function: 0.613222278887406
comprehension (big): 8.075551941990852
function (big): 16.2689924351871
```

Python 3.11.5

Reply:
Thanks @barakalon! I'll go ahead and revert this, then.